src/allotropy/parsers/moldev_softmax_pro/softmax_pro_parser.py

```diff
@@ -1,17 +1,266 @@
-from typing import Any
+from collections.abc import Iterator
+import uuid
 
+from allotropy.allotrope.models.plate_reader_benchling_2023_09_plate_reader import (
+    ContainerType,
+    DataSystemDocument,
+    DeviceSystemDocument,
+    FluorescencePointDetectionDeviceControlAggregateDocument,
+    FluorescencePointDetectionDeviceControlDocumentItem,
+    FluorescencePointDetectionMeasurementDocumentItems,
+    LuminescencePointDetectionDeviceControlAggregateDocument,
+    LuminescencePointDetectionDeviceControlDocumentItem,
+    LuminescencePointDetectionMeasurementDocumentItems,
+    MeasurementAggregateDocument,
+    Model,
+    PlateReaderAggregateDocument,
+    PlateReaderDocumentItem,
+    SampleDocument,
+    ScanPositionSettingPlateReader,
+    UltravioletAbsorbancePointDetectionDeviceControlAggregateDocument,
+    UltravioletAbsorbancePointDetectionDeviceControlDocumentItem,
+    UltravioletAbsorbancePointDetectionMeasurementDocumentItems,
+)
+from allotropy.allotrope.models.shared.definitions.custom import (
+    TQuantityValueDegreeCelsius,
+    TQuantityValueMilliAbsorbanceUnit,
+    TQuantityValueNanometer,
+    TQuantityValueNumber,
+    TRelativeFluorescenceUnit,
+    TRelativeLightUnit,
+)
+from allotropy.constants import ASM_CONVERTER_NAME, ASM_CONVERTER_VERSION
+from allotropy.exceptions import (
+    AllotropeConversionError,
+)
 from allotropy.named_file_contents import NamedFileContents
 from allotropy.parsers.lines_reader import CsvReader, read_to_lines
-from allotropy.parsers.moldev_softmax_pro.softmax_pro_structure import Data
+from allotropy.parsers.moldev_softmax_pro.softmax_pro_structure import (
+    Data,
+    PlateBlock,
+)
+from allotropy.parsers.utils.values import (
+    assert_not_none,
+)
 from allotropy.parsers.vendor_parser import VendorParser
 
+EPOCH = "1970-01-01T00:00:00-00:00"
+
 
 class SoftmaxproParser(VendorParser):
-    def to_allotrope(self, named_file_contents: NamedFileContents) -> Any:
+    def to_allotrope(self, named_file_contents: NamedFileContents) -> Model:
         lines = read_to_lines(named_file_contents.contents, encoding=None)
         reader = CsvReader(lines)
         data = Data.create(reader)
-        return self._get_model(data)
+        return self._get_model(named_file_contents.original_file_name, data)
+
+    def _get_model(self, file_name: str, data: Data) -> Model:
+        return Model(
+            field_asm_manifest="http://purl.allotrope.org/json-schemas/adm/plate-reader/BENCHLING/2023/09/plate-reader.schema",
+            plate_reader_aggregate_document=PlateReaderAggregateDocument(
+                device_system_document=DeviceSystemDocument(
+                    device_identifier="null",
+                    model_number="null",
```

Comment:
I really don't think we should be using a hardcoded "null" for either of these non-nullable identifiers. If we're saying that it's ok for a `DeviceSystemDocument` to be missing these fields, then they should allow `None` as an option. If it's not ok to be missing these fields for other uses of `DeviceSystemDocument`, then we may need a slightly different dataclass here.